src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -2666,7 +2660,7 @@ public void visitApply(JCMethodInvocation tree) {
                         && (tree.meth.hasTag(SELECT))
                         && ((JCFieldAccess)tree.meth).selected.hasTag(IDENT)
                         && TreeInfo.name(((JCFieldAccess)tree.meth).selected) == names._super;
-                boolean qualifierIsUniversal = allowUniversalTVars && qualifier.hasTag(TYPEVAR) && ((TypeVar)qualifier).universal;
+                boolean qualifierIsUniversal = allowUniversalTVars && qualifier.hasTag(TYPEVAR) && ((TypeVar)qualifier).isValueProjection();
```

Comment:
This should be `isUniversal()` I think - `wait` and friends should warn, even on `.ref` stuff.

Reply:
not sure to be honest, although there is no reference in the JEP to these warnings so we can do either now

Reply:
I double checked with Brian yesterday - he confirmed that both `T` and `T.ref` should not support identity operation (well, given warning when attempting to).

Reply:
ok, thanks for double checking, will fix this

Reply:
T.ref still support == null, but it's a corner case